Eatery/Controllers/Eateries/CampusMenuViewController.swift

```diff
@@ -0,0 +1,393 @@
+
+        mealViewController.view.layoutIfNeeded()
+        return mealViewController.tableView.contentSize.height
+    }
+    
+}
+
+private class CampusMenuInfoView: UIView {
+
+    private var statusLabel: UILabel!
+    private var hoursLabel: UILabel!
+    private var locationLabel: UILabel!
+    private var distanceLabel: UILabel!
+
+    var statusHero: HeroExtension<UILabel> {
+        return statusLabel.hero
+    }
+
+    var hoursHero: HeroExtension<UILabel> {
+        return hoursLabel.hero
+    }
+
+    var locationHero: HeroExtension<UILabel> {
+        return locationLabel.hero
+    }
+
+    var distanceHero: HeroExtension<UILabel> {
+        return distanceLabel.hero
+    }
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        statusLabel = UILabel()
+        statusLabel.isOpaque = false
+        statusLabel.textColor = .eateryBlue
+        statusLabel.font = .systemFont(ofSize: 14, weight: .semibold)
+        addSubview(statusLabel)
+        statusLabel.snp.makeConstraints { make in
+            make.top.equalToSuperview().inset(16)
+            make.leading.equalToSuperview().inset(16)
+        }
+
+        hoursLabel = UILabel()
+        hoursLabel.isOpaque = false
+        hoursLabel.textColor = .gray
+        hoursLabel.font = UIFont.systemFont(ofSize: 14, weight: .medium)
+        addSubview(hoursLabel)
+        hoursLabel.snp.makeConstraints { make in
+            make.centerY.equalTo(statusLabel)
+            make.leading.equalTo(statusLabel.snp.trailing).offset(2)
+        }
+
+        locationLabel = UILabel()
+        locationLabel.isOpaque = false
+        locationLabel.font = UIFont.systemFont(ofSize: 14, weight: .medium)
+        locationLabel.textColor = .gray
+        addSubview(locationLabel)
+        locationLabel.snp.makeConstraints { make in
+            make.leading.equalToSuperview().offset(16)
+            make.top.equalTo(statusLabel.snp.bottom).offset(8)
+            make.bottom.equalToSuperview().inset(16)
+        }
+
+        distanceLabel = UILabel()
+        distanceLabel.isOpaque = false
+        distanceLabel.textColor = .gray
+        distanceLabel.font = UIFont.systemFont(ofSize: 14, weight: .medium)
+        addSubview(distanceLabel)
+        distanceLabel.snp.makeConstraints { make in
+            make.leading.greaterThanOrEqualTo(hoursLabel.snp.trailing)
+            make.trailing.equalToSuperview().inset(16)
+            make.top.equalToSuperview().inset(16)
+        }
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func configure(presentation: EateryPresentation, address: String, distance: String) {
+        statusLabel.text = presentation.statusText
+        statusLabel.textColor = presentation.statusColor
+
+        hoursLabel.text = presentation.nextEventText
+
+        locationLabel.text = address
+
+        distanceLabel.text = distance
+    }
+
+}
+
+private class SeparatorView: UIView {
```

Comment:
same thing regarding moving this to its own file. seems like it could be reusable in the future